File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/LongTextFormatterFilter.java
package com.cedarsoft.commons.javafx;

import java.util.function.Predicate;

import javax.annotation.Nonnull;

import javafx.util.converter.NumberStringConverter;

/**
 * Processes text formatter changes. Ensures only valid longs are added
 */
class LongTextFormatterFilter extends NumberTextFormatterFilter {

  LongTextFormatterFilter(@Nonnull NumberStringConverter converter, @Nonnull Predicate<Long> filter) {
    super(converter, number -> filter.test(number.longValue()), 0);
  }

}

File: commons/javafx/src/main/kotlin/it/neckar/open/javafx/LongTextFormatterFilter.kt
package it.neckar.open.javafx

import javafx.util.converter.NumberStringConverter
import java.util.function.Predicate

/**
 * Processes text formatter changes. Ensures only valid longs are added
 */
internal class LongTextFormatterFilter(
  converter: NumberStringConverter,
  filter: Predicate<Long>,
) : NumberTextFormatterFilter(
  converter = converter,
  filter = Predicate { number: Number -> filter.test(number.toLong()) },
  emptyValue = 0
)
